### DISCUSSION TITLE
Can I train a model based on qwen:0.5b

### DISCUSSION THREAD

**Discussion by wisamidris7** [ggml-org/llama.cpp#8797] [General]
I searched everywhere and didn't find that could I retrain qwen:0.5b
For example to train him that his name is WSM Helper and to answer only accounting questions
And many many things and I need to train him locally without using any huggingface token cause I already pulled it from ollama

And some say use python and some c++ and I think that llama.cpp exe applications may contains one to retrain a model 

Help if anyone know how to retrain any model (without huggingface token).

**Comment by dspasyuk** [comment]
@wisamidris7 try using Llama-Factory https://github.com/hiyouga/LLaMA-Factory but you probably want to use embedding+vector database not retrained model.
